### ISSUE TITLE
HTTP ERROR 403 when creating a competition with AWS S3

### ISSUE THREAD

**Issue by sqhan-Husky** [codalab/codalab-competitions#3054]
Hi, 
noticed that you have update a new codalab these days, I mainly focus on the data storage and make lots of attempt.
Based on the old codalab instance on my own ec2, l deployed it successfully with Azure Blob and AWS S3, and failed on minio and another cloud storage, namely UCloud S3, which is compatible to AWS S3.

Having updated the codalab, l found it really weird that none of them have been successfully applied to create a competition (except for Azure, l haven't try it so far). l have already issued my problem on [minio](https://github.com/codalab/codalab-competitions/issues/3051) and [US3](https://github.com/codalab/codalab-competitions/issues/3040). 
Here, l want to report the current situation with AWS S3.

Your wiki on AWS storage is not updated, so I follow the settings of successfully-used s3 bucket on my old codalab instance (mainly the CORS and Principal) and create two new ones. Most of the settings in the `.env` remain unchangable. Here's the part of `.env`:
<img width="522" alt="1638456774(1)" src="https://user-images.githubusercontent.com/45597101/144445756-e1b1085d-e270-4161-87cb-cbcd35f45be0.png">
(I add region in the `AWS_S3_HOST` because if not, another problem will happen:
<img width="743" alt="image" src="https://user-images.githubusercontent.com/45597101/144447085-78b5eda7-4fdb-478b-93e0-f22d721a27e3.png">)

The current result is although it returns a `201` status code, the zip file is failed to save into my buckets. Then, the worker_site cannot unpack the zip with the location and return a `403` error:
![image](https://user-images.githubusercontent.com/45597101/144446925-ae6e3cd2-42fa-4138-b43a-015d6251a423.png)

l really thanks a lot for codalab, and hope my questions can be solved.

Best,
Siqi

**Comment by loicsarrazin** [comment]
hi @sqhan-Husky  
  
This should not change a whole lot of things but the only differences with my local env using S3 are this : 
S3_USE_SIGV4=True  
and the host ref not containing the region : 
AWS_S3_HOST=s3.amazonaws.com  
  
Maybe that could help. In the meantime I'll try to see if there is anything else I can help with with regards to CORS config.

**Comment by sqhan-Husky** [comment]
@loicsarrazin  Thank you for your response.

`S3_USE_SIGV4` you mentioned is in the `.env` in the old version of codalab and is replaced with `AWS_S3_SIGNATURE_VERSION` (l guess) in recent updated version.
Actually, l have experimented the two newly-created AWS buckets on my old codalab instance and it works well.

It indicates that my settings of buckets are correct and the problem is very likely to occur because of the updated `.env`.

**Comment by loicsarrazin** [comment]
Have you tried removing the region from `AWS_S3_HOST`?

**Comment by sqhan-Husky** [comment]
@loicsarrazin   I tried, of course.
If removing the region, the problem is:
![image](https://user-images.githubusercontent.com/45597101/144701214-0e56a87d-73b3-446c-b676-2dc56f84a6f8.png)

**Comment by Didayolo** [comment]
Any update on this?

**Comment by Didayolo** [comment]
@sqhan-Husky Did you update the repository `codalab-competitions` directly in a server that previously featured the Python2 version of CodaLab?

If so, you may need to re-build the Docker containers:

```bash
docker-compose build --no-cache
docker-compose up -d
```

**Comment by sqhan-Husky** [comment]
@Didayolo  I have updated the repository. and rebuild my container  Currently,  I use another S3 service now.   Thank you for your reply.

**Comment by Didayolo** [comment]
OK, thanks for the update. I guess I can close this issue then? Please re-open it if it is still useful.

**Comment by AyJayHam** [comment]
Hi, I am actually also facing issues when setting up Codalab with amazon s3 bucket.

I do get an 403 error as well. Digging into the requests I was able to find out that the reason i get this error is that Amazon throws. an error:` The request signature we calculated does not match the signature you provided. Check your key and signing method. - which comes with the 403. `

My `.env` file is set up correctly with the IAM user and the initial requests are successfully going through. But once Codalab tries to store the zip competition bundle file in the bucket the signing fails. 

My env:
![image](https://github.com/codalab/codalab-competitions/assets/41308456/b0b2a1b1-21ea-415e-a3e7-63b6726103a8)



It might have something todo with the boto3 being used.

Happy for any help!
